libs/books/data-access/src/lib/+state/reading-list.reducer.ts

```diff
@@ -53,11 +53,23 @@ const readingListReducer = createReducer(
   on(ReadingListActions.removeFromReadingList, (state, action) =>
     readingListAdapter.removeOne(action.item.bookId, state)
   ),
+  on(ReadingListActions.markBookAsRead, (state, action) => {
+    return readingListAdapter.updateOne({
+      id: action.book.bookId, changes: {
+        finished: true,
+        finishedDate: new Date().toISOString(),
```

Comment:
Reducers are pure functions ([https://ngrx.io/guide/store/reducers](https://ngrx.io/guide/store/reducers)). So, generating a date in a reducer would make it non-pure. I believe you should get the date elsewhere, and pass it in the reducer.

Reply:
I am passing the finished and finishedDate attributes through the store action when the user clicks on the button.